src/inmanta/ast/type.py

```diff
@@ -421,6 +546,23 @@ def type_string_internal(self) -> str:
     def get_location(self) -> None:
         return None
 
+    def corresponds_to(self, pytype: type[object]) -> bool:
+        if typing_inspect.is_union_type(pytype):
+            return False
+
+        if not typing_inspect.is_generic_type(pytype):
+            return issubclass(pytype, Sequence)
+
+        origin = typing_inspect.get_origin(pytype)
+        return issubclass(origin, Sequence)
+
+    def as_python_type_string(self) -> "str | None":
+        return "list[object]"
```

Comment:
```suggestion
        return "Sequence[object]"
```
Right?

Reply:
Same question for dict

Reply:
ostensibly yes, but then we also need the import or use is fully qualified.

it is a story of readability/clarity for less sophisticated users Vs precision for sophisticated users.

How would you solve the import problem? always print that in the example?

Reply:
> it is a story of readability/clarity for less sophisticated users Vs precision for sophisticated users.

In part yes. But there's also the aspect that a user might actually `append` elements to the list, which (I hope) would have no effect in the DSL. This would be an even worse experience for less sophisticated users imo.

> How would you solve the import problem? always print that in the example?

This is a good point, and I don't have a satisfying answer. We could check `if "Sequence" in dataclass_repr` etc, but that's not great either.

Reply:
Declaring it as Sequence will not stop anyone from appending to it, unless they have mypy configured and they pay attention to it. So while essentially a correct observation, I don't think it is a very strong argument.

Reply:
Ah yes, I was going off the assumption that mypy (and perhaps even mypy-baseline) would become part of the recommended (and thus conventional) development flow. Would that not be the case? If indeed my assumption is incorrect, then I have to agree that it's a weak argument, and that yours carry more weight.

I do want to make sure that we *allow* `Sequence`, but I think we already do.

Reply:
We definitely allow it. 

And even if we suggest using it, that is still a long shot from it being used.